src/DocChecks.jl

```diff
@@ -405,4 +408,38 @@ else
     footnotes(doc::Documents.Document) = nothing
 end
 
+
+# Link Checks.
+# ------------
+
+import Requests
+
+function linkcheck(doc::Documents.Document)
+    if doc.user.linkcheck
+        println(" > checking external URLs:")
+        for (src, page) in doc.internal.pages
+            println("   - ", src)
+            for element in page.elements
+                Walkers.walk(page.globals.meta, page.mapping[element]) do block
+                    linkcheck(block, doc)
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+function linkcheck(link::Base.Markdown.Link, doc::Documents.Document)
+    if !(link in doc.internal.locallinks)
+        local good = try
+            Requests.statuscode(Requests.head(link.url)) == 200
+        catch err
+            false
+        end
+        print_with_color(good ? :green : :red, good ? "     " : "   X ", link.url, "\n")
```

Comment:
Maybe one more character? In case there are a lot of status codes then it becomes a bit hard to read (e.g. if all links of a page fail etc.)
```
...
 > checking external URLs:
   - lib/internals/dom.md
   - lib/internals.md
   - lib/internals/builder.md
   - man/contributing.md
     200 http://juliacontrol.github.io/ControlSystems.jl/latest/
     200 https://juliafinance.github.io/Currencies.jl/latest/
   - lib/internals/writers.md
   - man/guide.md
      200 http://www.mkdocs.org/
      200 https://github.com/JuliaDocs/Documenter.jl/issues
   - lib/public.md
   - lib/internals/mdflatten.md
```

Also, I am wondering if, for consistency, we should also show the `200`s. Of course, in that case, if there's no color, there's nothing indicating whether the link was considered a success or not. OTOH, without it the user doesn't really know why the link was considered a success (unless they go and read the code and see that we're checking for the status code 200).